src/apple/system.rs

```diff
@@ -453,6 +453,28 @@ impl SystemExt for System {
     fn get_version(&self) -> Option<String> {
         get_system_info(libc::KERN_OSRELEASE, None)
     }
+
+    fn get_os_version(&self) -> Option<String> {
+        unsafe {
+            // get the size for the buffer first
+            let mut size = 0;
+            libc::sysctlbyname(
```

Comment:
There is `get_sys_value_by_name` just below. Please use it instead. ;)

Reply:
I did that on the second call, but for some reason on the first it wasn't setting the size for me when I pass in the null pointer, I'll try again :)

Reply:
Oh you're right, please also update `get_sys_value_by_name` so that it takes a `&mut` instead of a usize for the len otherwise you won't be able to use it.

Reply:
fixed in d3d6a3a